### PR TITLE
gitSync: add extraEnvFrom to git-sync containers to support proxy settings from k8s secret

### PR DESCRIPTION
**Pull request by rasulkarimov**
This PR introduces support for proxy settings in the gitSync containers to accommodate scenarios where the git repository is located behind a corporate proxy server. The proxy settings can be configured in a Kubernetes secret with keys HTTPS_PROXY, HTTP_PROXY, and NO_PROXY. These settings are then accessed via the 'dags.gitSync.proxySettingsSecret' variable, which provides them as environment variables for the gitSync containers.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by jedcunningham — CHANGES_REQUESTED** [review]
I'd much rather we move [gitSync.env](https://github.com/apache/airflow/blob/58820a914c9b84c378c9bc1acd4ddbcb4e4a3497/chart/values.schema.json#L7408) to accept [V1EnvVars](https://github.com/apache/airflow/blob/58820a914c9b84c378c9bc1acd4ddbcb4e4a3497/chart/values.schema.json#L8173-L8175) vs restricting it artificially like we are now.

@rasulkarimov, can you give that approach a shot instead?

**Comment by rasulkarimov** [comment]
@jedcunningham 
"restricting it artificially" - I agree with your point, I just wanted to follow the established pattern in the "git_sync_container" template. 
However, using gitSync.env is not ideal because proxy settings may contain secret credentials, which would require defining them directly in yaml.

Therefore, I have updated the PR to allow envFrom in git_sync_container. This will enable us to provide proxy settings with credentials to the git-sync container from a Kubernetes secret(without restricting it artificially).

**Review by romsharon98 — APPROVED** [review]

**Comment by romsharon98** [comment]
look good, can you add some tests?

**Comment by rasulkarimov** [comment]
> look good, can you fix current tests to validate your changes?

Thank you! Fixed tests.

**Review by romsharon98 — APPROVED** [review]

**Comment by rasulkarimov** [comment]
> I'd much rather we move [gitSync.env](https://github.com/apache/airflow/blob/58820a914c9b84c378c9bc1acd4ddbcb4e4a3497/chart/values.schema.json#L7408) to accept [V1EnvVars](https://github.com/apache/airflow/blob/58820a914c9b84c378c9bc1acd4ddbcb4e4a3497/chart/values.schema.json#L8173-L8175) vs restricting it artificially like we are now.
> 
> @rasulkarimov, can you give that approach a shot instead?

Can we merge this PR?

File: helm_tests/airflow_aux/test_pod_template_file.py [hunk 1]
```diff
@@ -98,6 +98,7 @@ def test_should_add_an_init_container_if_git_sync_is_true(self):
             "securityContext": {"runAsUser": 65533},
             "image": "test-registry/test-repo:test-tag",
             "imagePullPolicy": "Always",
+            "envFrom": [],
```
Comment (amoghrajesh) [inline helm_tests/airflow_aux/test_pod_template_file.py]: Can you also write tests where we override it?

**Review by amoghrajesh — COMMENTED** [review]
Also i think since it is a pretty trivial case, do we also document it?

File: helm_tests/airflow_aux/test_pod_template_file.py (hunk 1 above)
Comment (eladkal) [reply helm_tests/airflow_aux/test_pod_template_file.py]: @rasulkarimov can you address the above comment?

Comment (rasulkarimov) [reply helm_tests/airflow_aux/test_pod_template_file.py]: Updated the "extraEnvFrom" variables and "envFrom" asserts in helm_test.
Can someone trigger the test pipelines?

File: chart/templates/_helpers.yaml [hunk 2]
```diff
@@ -198,6 +198,7 @@ If release name contains chart name it will be used as a full name.
   image: {{ template "git_sync_image" . }}
   imagePullPolicy: {{ .Values.images.gitSync.pullPolicy }}
   securityContext: {{- include "localContainerSecurityContext" .Values.dags.gitSync | nindent 4 }}
+  envFrom: {{- include "custom_airflow_environment_from" . | default "\n  []" | indent 2 }}
```
Comment (jedcunningham) [inline chart/templates/_helpers.yaml]: We shouldn't be using the global `extraEnvFrom` - that's intended for Airflow containers, which gitsync isn't.

Can you define a separate `dags.gitSync.envFrom`?

Comment (rasulkarimov) [reply chart/templates/_helpers.yaml]: Defined separate variable for gitSync envFrom - dags.gitSync.envFrom

Comment (rasulkarimov) [reply chart/templates/_helpers.yaml]: @jedcunningham  can we merge this PR?

**Comment by rasulkarimov** [comment]
> I'd much rather we move [gitSync.env](https://github.com/apache/airflow/blob/58820a914c9b84c378c9bc1acd4ddbcb4e4a3497/chart/values.schema.json#L7408) to accept [V1EnvVars](https://github.com/apache/airflow/blob/58820a914c9b84c378c9bc1acd4ddbcb4e4a3497/chart/values.schema.json#L8173-L8175) vs restricting it artificially like we are now.
> 
> @rasulkarimov, can you give that approach a shot instead?

can we merge?

**Comment by eladkal** [comment]
cc @jedcunningham as you marked request changes

**Review by jedcunningham — APPROVED** [review]

**Comment by jedcunningham** [comment]
Thanks @rasulkarimov! Sorry for the slow merge. Congrats on your first commit 🎉

**Pull request merged**